src/fastoad/model_base/flight_point.py

```diff
@@ -252,19 +382,38 @@ def remove_field(cls, name):
         :param name: field name
         """
         if name in cls.__annotations__:
+            cls._redeclare_fields()
             del cls.__init__  # Delete constructor to allow it being rebuilt with dataclass() call
             delattr(cls, name)
             del cls.__annotations__[name]
             dataclass(cls)
-            if name in cls._units:
-                del cls._units[name]
 
-    def scalarize(self):
+    @classmethod
+    def _get_field_descriptors(cls) -> Dict[str, _FieldDescriptor]:
         """
-        Convenience method for converting to scalars all fields that have a
-        one-item array-like value.
+        Uses this method instead of accessing cls.__field_descriptors to ensure it
+        will always be correctly populated.
         """
-        self_as_dict = asdict(self)
-        for field_name, value in self_as_dict.items():
-            if np.size(value) == 1:
-                setattr(self, field_name, np.asarray(value).item())
+        if not cls.__field_descriptors:
+            cls.__field_descriptors = {
+                cls_field.name: cls_field.metadata[FIELD_DESCRIPTOR]
+                for cls_field in fields(cls)
+                if not cls_field.name.startswith("_")
+            }
+
+        return cls.__field_descriptors
+
+    @classmethod
+    def _redeclare_fields(cls):
+        """
+        To be done before "re-dataclassing" cls.
+
+        In current state, fields are now declared using only default value (no metadata). We have
+        to redo the field declaration properly.
+        """
+        for cls_field in fields(cls):
+            setattr(
+                cls, cls_field.name, field(default=cls_field.default, metadata=cls_field.metadata)
+            )
+
+        cls.__field_descriptors = {}  # Will need to rebuild this dict on next usage.
```

Comment:
I don't really understand why this method (and more generally, the deletion of the constructor and other attributes) is necessary. would you mind explaining it to me ?

Reply:
Actually, we have to hack a little the dataclass mechanism for allowing the extensibility of `FlightPoint`.
Dataclasses are not designed for later addition of fields, so I had to do trial and error/debug to have it working.

Dataclasses generate their own `__init__`, and I found that a second "`dataclass` decoration" (i.e. calling `dataclass(cls)` is not enough for having an entirely new `__init__`).

I also found that after `__init__` generation, the class attributes, initially declared with `field(...)` become declared with only the default value (data in `field(...)` are transferred in `Field` instances we can access through `dataclasses.fields()`). So I had to redo a proper declaration to avoid losing all previously declared metadata during the next "`dataclass` decoration".

If you want to better understand what's going on, do not hesitate to play with the code. Removing some line and running the tests should show you the reason for the line (and if all runs smoothly, I'm interested in knowing that 😉).